robot/impl/local_robot_test.go

```diff
@@ -3359,3 +3360,191 @@ func TestCloudMetadata(t *testing.T) {
 		})
 	})
 }
+
+func TestReconfigureOnModuleRename(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Precompile complex module to avoid timeout issues when building takes too long.
+	complexPath := rtestutils.BuildTempModule(t, "examples/customresources/demos/complexmodule")
+
+	// Manually define mybase model, as importing it can cause double registration.
+	myBaseModel := resource.NewModel("acme", "demo", "mybase")
+
+	// Create config with at least one module and a component from that module
+	cfg := &config.Config{
+		Modules: []config.Module{
+			{
+				Name:     "mod",
+				ExePath:  complexPath,
+				LogLevel: "info",
+			},
+		},
+		Components: []resource.Config{
+			{
+				Name:  "myBase",
+				API:   base.API,
+				Model: myBaseModel,
+				Attributes: rutils.AttributeMap{
+					"motorL": "motor1",
+					"motorR": "motor2",
+				},
+			},
+			{
+				Name:                "motor1",
+				API:                 motor.API,
+				Model:               fakeModel,
+				ConvertedAttributes: &fakemotor.Config{},
+			},
+			{
+				Name:                "motor2",
+				API:                 motor.API,
+				Model:               fakeModel,
+				ConvertedAttributes: &fakemotor.Config{},
+			},
+		},
+	}
+
+	// Create copy of cfg since Reconfigure (called when setting up a robot) modifies cfg.
+	cfgCopy := *cfg
+	r := setupLocalRobot(t, ctx, cfg, logger)
+
+	// Use copy to modify config by renaming module
+	cfgCopy.Modules[0].Name = "mod-renamed"
+
+	// Create copy of cfgCopy to test against
+	expectedCfg := cfgCopy
+	r.Reconfigure(ctx, &cfgCopy)
+
+	actualCfg := r.Config()
+	robotResources := r.ResourceNames()
+
+	// Verify attributes of resources and config
+
+	// Manually inspect config components
+	test.That(t, len(actualCfg.Components), test.ShouldEqual, 3)
+	for _, comp := range actualCfg.Components {
+		isMyBase := comp.Equals(cfg.Components[0])
+		isMotor1 := comp.Equals(cfg.Components[1])
+		isMotor2 := comp.Equals(cfg.Components[2])
+		test.That(t, isMyBase || isMotor1 || isMotor2, test.ShouldBeTrue)
+	}
+
+	// Manually inspect module
+	test.That(t, len(actualCfg.Modules), test.ShouldEqual, 1)
+	test.That(t, actualCfg.Modules[0].Equals(expectedCfg.Modules[0]), test.ShouldBeTrue)
+
+	// Remove default services
+	defaultSvcs := resource.DefaultServices()
+	test.That(t, len(defaultSvcs), test.ShouldEqual, 2)
+	robotResources = slices.DeleteFunc(robotResources, func(resource resource.Name) bool {
+		return slices.Contains(defaultSvcs, resource)
+	})
+
+	// Manually inspect resources to ensure they all get built
+	test.That(t, len(robotResources), test.ShouldEqual, 3)
+	for _, resource := range robotResources {
```

Comment:
use a [helper](https://github.com/viamrobotics/rdk/blob/main/robot/impl/robot_reconfigure_test.go#L148-L154) for this so you don't have to remove service names and can also use `cfg.Components[0].ResourceName()` to get the matching resource name